test/IntegrationTests/EntityFrameworkCoreTests.cs

```diff
@@ -15,18 +16,43 @@ public EntityFrameworkCoreTests(ITestOutputHelper output)
     {
     }
 
+    public static IEnumerable<object[]> GetData()
+    {
+        foreach (var version in LibraryVersion.EntityFrameworkCore)
+        {
+            yield return new[] { version[0], true };
+            yield return new[] { version[0], false };
+        }
+    }
+
     [Theory]
     [Trait("Category", "EndToEnd")]
-    [MemberData(nameof(LibraryVersion.EntityFrameworkCore), MemberType = typeof(LibraryVersion))]
-    public void SubmitsTraces(string packageVersion)
+    [MemberData(nameof(GetData))]
+    public void SubmitTraces(string packageVersion, bool dbStatementForText)
     {
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_ENTITYFRAMEWORKCORE_SET_DBSTATEMENT_FOR_TEXT", dbStatementForText.ToString());
         using var collector = new MockSpansCollector(Output);
         SetExporter(collector);
-        collector.Expect("OpenTelemetry.Instrumentation.EntityFrameworkCore");
+        collector.Expect("OpenTelemetry.Instrumentation.EntityFrameworkCore", span => Expect(span, dbStatementForText));
```

Comment:
I find `private static bool Expect(Span span, bool dbStatementForText)` too complex and I also do not like the name. I think having the logic inline is easier to follow.

```suggestion
        if (dbStatementForText)
        {
            collector.Expect("OpenTelemetry.Instrumentation.EntityFrameworkCore", span => span.Attributes.Any(attr => attr.Key == "db.statement" && !string.IsNullOrWhiteSpace(attr.Value?.StringValue)));
        }
        else
            collector.Expect("OpenTelemetry.Instrumentation.EntityFrameworkCore", span => !span.Attributes.Any(attr => attr.Key == "db.statement"));
        }
```

Reply:
6a0fea137be9334f77449cd3d7c0cf36d5ef2f79